test/provider_os_memory.cpp

```diff
@@ -39,396 +47,243 @@ static const char *Native_error_str[] = {
-    const char *message;
-    int32_t error;
-    umfMemoryProviderGetLastNativeError(os_memory_provider, &message, &error);
-    ASSERT_EQ(error, UMF_OS_RESULT_ERROR_FREE_FAILED);
-    ASSERT_EQ(compare_native_error_str(message, error), 0);
+TEST_P(umfProviderTest, provider_os_memory_purge_lazy) {
+    test_alloc_free_success(provider.get(), SIZE_4100, 0, PURGE_LAZY);
+}
 
-    umfMemoryProviderDestroy(os_memory_provider);
+TEST_P(umfProviderTest, provider_os_memory_purge_force) {
+    test_alloc_free_success(provider.get(), SIZE_4100, 0, PURGE_FORCE);
 }
 
-TEST_F(test, provider_os_memory_get_min_page_size) {
-    umf_result_t umf_result;
-    umf_memory_provider_handle_t os_memory_provider = nullptr;
-    umf_os_memory_provider_params_t os_memory_provider_params =
-        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+// negative tests using test_alloc_failure
 
-    umf_result = umfMemoryProviderCreate(&UMF_OS_MEMORY_PROVIDER_OPS,
-                                         &os_memory_provider_params,
-                                         &os_memory_provider);
-    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
-    ASSERT_NE(os_memory_provider, nullptr);
+TEST_P(umfProviderTest, provider_os_memory_alloc_4100_align_page_minus_1) {
+    test_alloc_failure(provider.get(), SIZE_4100, page_size - 1,
+                       UMF_RESULT_ERROR_INVALID_ARGUMENT, 0);
+}
 
-    size_t min_page_size;
-    umf_result = umfMemoryProviderGetMinPageSize(os_memory_provider, nullptr,
-                                                 &min_page_size);
-    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
-    ASSERT_LE(min_page_size, SIZE_4K);
+TEST_P(umfProviderTest, provider_os_memory_alloc_4100_align_one_half_pages) {
+    test_alloc_failure(provider.get(), SIZE_4100, page_size + (page_size / 2),
+                       UMF_RESULT_ERROR_INVALID_ARGUMENT, 0);
+}
 
-    umfMemoryProviderDestroy(os_memory_provider);
+TEST_P(umfProviderTest, provider_os_memory_alloc_MMAP_FAILED) {
+    test_alloc_failure(provider.get(), -1, 0,
+                       UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC,
+                       UMF_OS_RESULT_ERROR_ALLOC_FAILED);
 }
 
-TEST_F(test, provider_os_memory_get_recommended_page_size) {
-    umf_result_t umf_result;
-    umf_memory_provider_handle_t os_memory_provider = nullptr;
-    umf_os_memory_provider_params_t os_memory_provider_params =
-        UMF_OS_MEMORY_PROVIDER_PARAMS_TEST;
+// other positive tests
 
-    umf_result = umfMemoryProviderCreate(&UMF_OS_MEMORY_PROVIDER_OPS,
-                                         &os_memory_provider_params,
-                                         &os_memory_provider);
+TEST_P(umfProviderTest, provider_os_memory_get_min_page_size) {
+    size_t min_page_size;
+    umf_result_t umf_result = umfMemoryProviderGetMinPageSize(
+        provider.get(), nullptr, &min_page_size);
     ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
-    ASSERT_NE(os_memory_provider, nullptr);
+    ASSERT_LE(min_page_size, SIZE_4100);
```

Comment:
Why compare with 4100? Should we even have a hardcoded page size here? There can be some systems with different page sizes right?

Reply:
Done